int32 nextNode
string action
